### ISSUE TITLE
Support matrix

### ISSUE THREAD

**Issue by igorpecovnik** [Miouyouyou/armbian-desktop#13]
- when we want to install a desktop, that is only aarch64, we need to throw out an error
- in case one selects gnome for 32bit we show  a warning (if it works at all on 32bit?)
- the same for Debian / Ubuntu

**Comment by Miouyouyou** [comment]
* Hmm... Are some desktop unsupported on armhf ? Maybe some "custom/arch" folder might be needed, to only show architecture specific desktop for these architectures. That way the desktop "would not exist" for other architectures, and then we can throw an error if a non-existing desktop is chosen.  
Same thing for appgroups.
* Again... if it's not supported at all, the packages won't be available so maybe we should just avoid providing this option for these architectures.

Anyway, yeah, an error should be thrown if unavailable desktop / appgroups are chosen, to avoid any confusion.

**Comment by igorpecovnik** [comment]
Some desktop doesn't have support for armhf - at least binary packages doesn't exits (dde for example). Perhaps reverse logic? Not sure atm, just we need to solve this somehow.

**Comment by Miouyouyou** [comment]
Yeah. I guess we'll need some arch specific folder. I need to think about it a little more.

**Comment by igorpecovnik** [comment]
also in this context we need DE support status like we have for userland. We will officially only support A and B desktop, others will be in "not supported" status. On the desktop names, perhaps also shorting to DE and showing status in ()

xfce | Xfce desktop enfironment -> xfce | Xfce DE (supported)
cinanamon | Cinanamon desktop enfironment -> cinanamon | Cinanamon DE (CSC)

**Comment by Miouyouyou** [comment]
Alright, I've added support for package.remove. I'll try to squeeze off arch limitations today.

I'm thinking about which way would be better. My best guess would be a file named "only_for" inside desktop environments and appgroups.

If, and only if, the file is present, its content is grep'ed for the current targeted architecture. If the architecture is not found (grep returned nothing), the option is not provided in the menus.
Then the same logic can be applied to generate errors, if an invalid option is selected.

**Comment by Miouyouyou** [comment]
Ok, preliminary support has been added (much later though) : https://github.com/Miouyouyou/armbian-desktop/commit/e9c7a34e6502484af374e4f58ff1baa10cfa27ff

I'll see how to generalize this for appgroups, and then anything just in case "cli appgroups" were to be added afterwards.
